internal/infrastructure/router/public.go

```diff
@@ -69,7 +70,27 @@ func newRbacSkipper(service string) echo_middleware.Skipper {
 		panic("service is an empty string")
 	}
 	return func(c echo.Context) bool {
-		return c.Path() == "/api/"+service+"/v1/openapi.json"
+		ctx := c.Request().Context()
+		// The access to the openapi specification is public
+		if c.Path() == "/api/"+service+"/v1/openapi.json" {
+			slog.DebugContext(ctx, "Skipping "+c.Path())
+			return true
+		}
+		cc, ok := c.(middleware.DomainContextInterface)
+		if !ok {
+			slog.WarnContext(ctx, "'c' is not a DomainContextInterface")
+			return false
+		}
+		if cc.XRHID().Identity.Type == "System" {
```

Comment:
Is the routes check needed? Given that `enforce identity` middlewares check if the right identity type is present for various routes, we could use something like this:
```go
		cc, ok := c.(middleware.DomainContextInterface)
		if !ok {
			slog.DebugContext(ctx, "RBAC: not a DomainContextInterface")
			return true
		}

		// RBAC is applicable only for path that have RH ID of type User
		identity := cc.XRHID()
		if identity == nil {
			slog.DebugContext(ctx, "RBAC: XRHID is not set")
			return true
		}
		if identity.Identity.Type != "User" {
			slog.DebugContext(ctx, "RBAC: XRHID is not a User")
			return true
		}
```

Btw, I' m not logging with word "skipping" and the "path" as it is logged in the `rbac.go`. 

Side-note, I wonder what `type` has service account. Looking into https://github.com/RedHatInsights/platform-go-middlewares/blob/master/identity/identity.go#L55 , there seems to be 5 structs that look as identities. 

If service account has a different type, then `userIdentityMiddleware` and `mixedIdentityMiddleware` would need to be enhanced with a new predicate and by code here with an additional type. Or it can be

```
		if identity.Identity.Type == "System" {
			slog.DebugContext(ctx, "RBAC: XRHID has type 'System'")
			return true
		}
```

Reply:
Thanks for the suggestion.

Yes, enforcement should warranty that `XRHID` is valid and store the unserialized value in the context. I would add the change :+1:

About the route check, I am afraid it is necessary.

Reply:
I step back; you are right, the route check is not necessary at the rbac skipper. I will add it :+1:  thanks